rulesets/java/jpinpoint-rules.xml

```diff
@@ -384,7 +384,7 @@ class Good {
     ]/(ClassType/@SimpleName|PrimitiveType/@Kind)
     or
     (: or its classType is equal to the field classtype :)
-    (ClassType/@SimpleName|PrimitiveType/@Kind) = ancestor::ClassBody//FieldDeclaration[
+    (ClassType/@SimpleName|PrimitiveType/@Kind) = ancestor::ClassBody[1]/FieldDeclaration[
```

Comment:
replacing the `//` with `[1]/` is ok, so FieldDeclaration is always _direct_ parent?

Reply:
As far as I know, it is always a direct child.

Reply:
Assuming it works the same for all. Tested for #576, has unit tests

Reply:
ok, let's try it with some runs with IntelliJ PMD after merge :-)